storybookjs/storybook
# CSF3: Add title prefix support for stories with custom titles

Issue: #17508 

## What I did
I changed the autoTItle function to userOrAutoTitle, which enables inputting a custom title. If a title is given and the import path matcher, matches the file name, it will add a prefix if available.

If no prefix is available the method will just return the title.

If no title is given, the default autoTitle behaviour is used (returning undefined if no path is matched).

## How to test
Unit tests have been added.

Testing can be done by creating some stories with a custom title, and in main.js specifying a title prefix for files in the directory of the created story. Then check if the prefixes are correctly applied. It's also possible to check if autoTitle still works.



If your answer is yes to any of these, please make sure to include it in your PR.
